iothub/service/src/Registry/DevicesClient.cs

```diff
@@ -663,26 +663,115 @@ public virtual async Task<JobProperties> GetJobAsync(string jobId, CancellationT
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        public virtual async Task<ExportJobProperties> GetExportJobAsync(string jobId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Getting job {jobId}", nameof(GetExportJobAsync));
+
+            Argument.AssertNotNullOrWhiteSpace(jobId, nameof(jobId));
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Get, GetJobUri(jobId), _credentialProvider);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                await HttpMessageHelper.ValidateHttpResponseStatusAsync(HttpStatusCode.OK, response).ConfigureAwait(false);
+                return await HttpMessageHelper.DeserializeResponseAsync<ExportJobProperties>(response).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"Getting job {jobId} threw an exception: {ex}", nameof(GetExportJobAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Getting job {jobId}", nameof(GetExportJobAsync));
+            }
+        }
+
+        /// <summary>
+        /// List all registry import jobs for the IoT hub.
+        /// </summary>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>IEnumerable of ImportJobProperties of all jobs for this IoT hub.</returns>
+        /// <exception cref="IotHubServiceException">
+        /// If IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubServiceException"/> with <see cref="IotHubErrorCode.ThrottlingException"/> is thrown.
+        /// For a complete list of possible error cases, see <see cref="IotHubErrorCode"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        public virtual async Task<IEnumerable<ImportJobProperties>> GetImportJobsAsync(CancellationToken cancellationToken = default)
```

Comment:
Do we need separate methods to get all import jobs and all export jobs?

Reply:
That was one of the snags of splitting them up. You can't return just one of the properties, but maybe they only need some common properties back that have to do with status, and not include the parameters originally passed.

This might require yet another type.